### PR TITLE
Add helper text for external links

### PR DESCRIPTION
**Pull request by lcollette**
Adds helper text for screen readers only if external link icon is displayed.

#### Changes proposed in this Pull Request

If the external link icon is shown, add helper text for screen readers. 

#### Testing instructions

1. Switch to this PR.
2. Visit https://wordpress.com/following/manage
3. Inspect any link with icon. You will see the helper text present but not visible. 
4. Go to https://wordpress.com/read/feeds/25823/posts/2317223963
5. Inspect the article title. The helper text is not shown.

Fixes #

### CODE REVIEW

**Comment by donpark** [comment]
Patch worked as expected but CircleCI test is failing in `test-client`:
```
 FAIL  client/my-sites/email/email-management/gsuite-user-item/test/index.js
  ● GSuiteUserItem › renders user item with manage correctly

    expect(received).toMatchSnapshot()

    Snapshot name: `GSuiteUserItem renders user item with manage correctly 1`

    - Snapshot
    + Received

    @@ -23,7 +23,12 @@
              <path
                d="M19 13v6c0 1.105-.895 2-2 2H5c-1.105 0-2-.895-2-2V7c0-1.105.895-2 2-2h6v2H5v12h12v-6h2zM13 3v2h4.586l-7.793 7.793 1.414 1.414L19 6.414V11h2V3h-8z"
              />
            </g>
          </svg>
    +     <span
    +       className="screen-reader-text"
    +     >
    +       (opens in a new tab)
    +     </span>
        </a>
      </li>

      23 | 			)
      24 | 			.toJSON();
    > 25 | 		expect( tree ).toMatchSnapshot();
         | 		               ^
      26 | 	} );
      27 | 
      28 | 	test( 'renders user item finish setup button correctly', () => {

      at Object.toMatchSnapshot (client/my-sites/email/email-management/gsuite-user-item/test/index.js:25:18)

 › 1 snapshot failed.
```

Also, it'd be nice to have original issue linked to this PR using `Fixes #` part of the PR description.

**Comment by lcollette** [comment]
> Also, it'd be nice to have original issue linked to this PR using Fixes # part of the PR description.

There isn't one. The PR serves as the issue in this case.

File: client/components/external-link/index.jsx [hunk 1]
```diff
@@ -73,9 +73,17 @@ class ExternalLink extends Component {
 				{ this.props.icon && this.props.showIconFirst && iconComponent }
 				{ this.props.children }
 				{ this.props.icon && ! this.props.showIconFirst && iconComponent }
+				{ this.props.icon && (
+					// eslint-disable-next-line wpcalypso/jsx-classname-namespace
+					<span className="screen-reader-text">
+						{
+							/* translators: accessibility text */
+							'(opens in a new tab)'
```
Comment (simison) [inline client/components/external-link/index.jsx]: This needs to be wrapped in `translate( '(opens in a new tab)' )` or otherwise it won't be translated.

Comment (donpark) [reply client/components/external-link/index.jsx]: my bad. I missed that while reviewing. thx for spotting.

@lcollette can you address this issue when you can? thx

Comment (lcollette) [reply client/components/external-link/index.jsx]: Got it!

Comment (simison) [reply client/components/external-link/index.jsx]: No worries, happens. It'll be a simple fix. :-)

**Pull request merged**